### PR TITLE
[DROOLS-5823] executable-model test failure in test-compiler-integrat…

### PR DESCRIPTION
**Pull request by tkobayas**
…ion NamedConsequencesTest

- Introducing EdgeCaseNonExecModelTest

**JIRA**: 
https://issues.redhat.com/browse/DROOLS-5823

Most of issues are fixed by JIRAs linked to https://issues.redhat.com/browse/DROOLS-5823

Remaining issue ( DROOLS-6290 , https://github.com/kiegroup/drools/pull/3634 ) is considered as a minor edge case and it will likely require large changes so isolated to EdgeCaseNonExecModelTest to be revisited. 

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: drools-test-coverage/test-compiler-integration/src/test/java/org/drools/mvel/integrationtests/NamedConsequencesTest.java [hunk 1]
```diff
@@ -225,6 +219,7 @@ public void testNamedConsequencesOutsideOR() {
         assertTrue( results.contains( "stilton" ) );
     }
 
+    @Ignore("Moved to EdgeCaseNonExecModelTest")
```
Comment (tkobayas) [inline drools-test-coverage/test-compiler-integration/src/test/java/org/drools/mvel/integrationtests/NamedConsequencesTest.java:222]: I just put `@Ignore` here. So the test methods are duplicated in the original class `NamedConsequencesTest`  and `EdgeCaseNonExecModelTest`. But I think it's easier for developers to find that the test exists in `NamedConsequencesTest` and re-enable the test when fixed. I'm open to other approaches (e.g. just remove from `NamedConsequencesTest`).

**Comment by tkobayas** [comment]
Hi @mariofusco @lucamolteni @hellowdan @pibizza 

Currently we are working on exec-model issues which are linked from https://issues.redhat.com/browse/DROOLS-5729

Sometimes we can simply fix all issues in one test class.

But sometimes one of issues might be a very minor edge case while it may require large changes (e.g. https://github.com/kiegroup/drools/pull/3634  -- no rushing at all, Mario :) ). In that case, I think we may isolate such a test method to a special place (= `EdgeCaseNonExecModelTest`) so we will revisit later.

pros:
- By isolating the test method, the original test class can run with both non-exec-model and exec-model. So wider coverage is ensured.
- Accumulating such edge cases in `EdgeCaseNonExecModelTest`, we can easily go through such unfixed cases so it will help prioritizing in the future.

Do you think it's good to go?

File: drools-test-coverage/test-compiler-integration/src/test/java/org/drools/mvel/integrationtests/NamedConsequencesTest.java (hunk 1 above)
Comment (hellowdan) [reply drools-test-coverage/test-compiler-integration/src/test/java/org/drools/mvel/integrationtests/NamedConsequencesTest.java:222]: I see no problem with this approach. Maybe @pibizza has a different idea?

**Review by hellowdan — APPROVED** [review]

File: drools-test-coverage/test-compiler-integration/src/test/java/org/drools/mvel/integrationtests/EdgeCaseNonExecModelTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                     "then[t1]\n" +
+                     "    results.add( $a.getType() );\n" +
+                     "end\n";
+
+        List<String> results = executeTestWithDRL(str);
+
+        assertEquals(2, results.size());
+        assertTrue(results.contains("cheddar"));
+        assertTrue(results.contains("stilton"));
+    }
```
Comment (pibizza) [inline drools-test-coverage/test-compiler-integration/src/test/java/org/drools/mvel/integrationtests/EdgeCaseNonExecModelTest.java:76]: Can we replace it with:

 assertThat(results).containsExactlyInAnyOrder("cheddar", "stilton");

There are a number of cases of this in drools code - not just this test.

Comment (pibizza) [reply drools-test-coverage/test-compiler-integration/src/test/java/org/drools/mvel/integrationtests/EdgeCaseNonExecModelTest.java:76]: @hellowdan @tkobayas The approach seems correct. Separating special cases from normal ones also make tests easier to read.

**Review by mariofusco — APPROVED** [review]
